v1/storage/inmem/txn.go

```diff
@@ -118,19 +130,39 @@ func (txn *transaction) Write(op storage.PatchOp, path storage.Path, value any)
 	return nil
 }
 
+func comparableEquals(a, b any) bool {
+	switch a := a.(type) {
+	case nil:
+		return b == nil
+	case bool:
+		if vb, ok := b.(bool); ok {
+			return vb == a
+		}
+	case string:
+		if vs, ok := b.(string); ok {
+			return vs == a
+		}
+	case json.Number:
```

Comment:
At this point, we know that we can't possible be comparing an `int` and a `json.Number`?

Reply:
I guess that can happen if the client says “don’t roundtrip on write” and then goes on to still send data that needs to be roundtripped. Then this would just return false, right?